hugolib/page.go

```diff
@@ -1115,215 +1115,213 @@ func (p *Page) prepareForRender(cfg *BuildCfg) error {
 var ErrHasDraftAndPublished = errors.New("both draft and published parameters were found in page's frontmatter")
 
 func (p *Page) update(f interface{}) error {
-	if f == nil {
-		return errors.New("no metadata found")
-	}
-	m := f.(map[string]interface{})
-	// Needed for case insensitive fetching of params values
-	helpers.ToLowerMap(m)
-
 	var modified time.Time
-
-	var err error
 	var draft, published, isCJKLanguage *bool
-	for k, v := range m {
-		loki := strings.ToLower(k)
-		switch loki {
```

Comment:
This switch needs to be on the same indentation level as before (first level). So, either return early (preferred), or if that is not possible for some reason: Extract method or func.

Reply:
Other than that it looks good -- but see the commit message guidelines.

Reply:
Returning early defeats the purpose, as some of the logic at the end of the function must execute whether or not there is front matter. So I tried refactoring this function into two: processFrontmatterMetadata and processDerivedMetadata. But the problem is that the CJKLanguage detection logic depends on knowledge of what happened up top (hence the special treatment with the *isCJKLanguage var).

```
	if isCJKLanguage != nil {
		p.isCJKLanguage = *isCJKLanguage
	} else if p.s.Cfg.GetBool("hasCJKLanguage") {
		if cjk.Match(p.rawContent) {
			p.isCJKLanguage = true
		} else {
			p.isCJKLanguage = false
		}
	}
	p.params["iscjklanguage"] = p.isCJKLanguage
```

So the extraction is possible, but quite ugly.

Reply:
Ok I just realized there is another approach: always pass in a frontmatter map to this method, an empty one in the case that there is no front matter. Does that work for you? Essentially, which do you prefer:

1. Keep the big monolithic function we have now (switch restored to first level), which more easily allows for coupled logic as we have now, or,
2. Refactor it into processFrontmatterMetadata and processDerivedMetadata functions, but with some strange (looking) args such as `didFrontMatterSpecifyIsCJK`.

Reply:
Empty map in works, and kind of matches the method signuature.